src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -173,15 +173,14 @@ def test_input_equals_output(resource_client, input_model, output_model):
     # only comparing properties in input model to those in output model and
     # ignoring extraneous properties that maybe present in output model.
     try:
-        resource_client.transform_model(pruned_input_model, pruned_output_model)
         for key in pruned_input_model:
             if key in resource_client.properties_without_insertion_order:
                 assert test_unordered_list_match(
                     pruned_input_model[key], pruned_output_model[key]
                 )
             else:
-                assert resource_client.compare(
-                    pruned_input_model[key], pruned_output_model[key]
+                assert (
+                    pruned_input_model[key] == pruned_output_model[key]
```

Comment:
working on a new change for this line. To check for comparison on nested properties